hack/dev-setup-vagrant

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+#
+# Copyright 2018 The Gardener Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DEV_DIR=$(dirname "${0}")/../dev
+EXAMPLE_DIR=$(dirname "${0}")/../example
+MINIKUBE_IP=$(minikube ip)
+IP_ROUTE=$(ip route get 1)
+IP_ADDRESS=$(echo ${IP_ROUTE#*src} | awk '{print $1}')
+
+kubectl apply -f ${EXAMPLE_DIR}/cloudprofile-vagrant.yaml
+kubectl apply -f ${EXAMPLE_DIR}/secret-dev-vagrant.yaml
+kubectl apply -f ${EXAMPLE_DIR}/privatesecretbinding-core-vagrant.yaml
+
+kubectl config set-cluster gardener-dev \
+  --kubeconfig ${DEV_DIR}/vagrant-kubeconfig \
```

Comment:
Can we call `vagrant-kubeconfig` like `vagrant-seed-kubeconfig` or `minikube-seed-kubeconfig` or something elsel; just to make it clearer what it is?